### ISSUE TITLE
Adding User to Org fails

### ISSUE THREAD

**Issue by JonasDev17** [fschick/Keycloak.RestApiClient#11]
When trying to add an existing user to an org with the function PostOrganizationsMembersByIdAsync, I always get the error that the user does not exist. I checked whether the endpoint works by looking at the network tab in my browser when using the Keycloak UI and I tried it using Postman and both work but something's different when using this client. 

What I suspect is that the body is sent differently than how Keycloak expects it. Keycloak expects just an Id in the body but the type has to be application/json (basically invalid json)

![image](https://github.com/user-attachments/assets/d78b87e0-098b-41af-b7a5-b721aab660cb)

![image](https://github.com/user-attachments/assets/583a1e35-02af-4341-9d72-a3cf99a86174)

**Comment by JonasDev17** [comment]
I've done some more debugging but it does not look like the body is sent in a wrong format. The OpenAPI client takes the string can converts it to bytes (utf-8): `httpRequestMessage.Content = new StringContent(customJsonCodec.Serialize(options.Data), new UTF8Encoding(), "application/json");` I did decode the bytes back to string and got the Guid I sent.
Not sure whats wrong...

**Comment by JonasDev17** [comment]
Ok I figured it out, It adds double quotes at the start and end of the Id.

**Comment by fschick** [comment]
Do you mean somethings like this `{ "Id": "bc724c54-618f-4455-9f79-382425b051c8", ... }`? This is valid JSON, isn't it?

**Comment by JonasDev17** [comment]
> Do you mean somethings like this `{ "Id": "bc724c54-618f-4455-9f79-382425b051c8", ... }`? This is valid JSON, isn't it?

No, the RestApiClient sends it as:

`"bc724c54-618f-4455-9f79-382425b051c8"`

while it should be sent as:

`bc724c54-618f-4455-9f79-382425b051c8`

I figured this out by debugging:

![image](https://github.com/user-attachments/assets/0a53d9c0-03f0-4405-ae7a-2a7fc6861728)

this is the binary content that is sent and when decoding it results in:

![image](https://github.com/user-attachments/assets/7b3b6128-b032-484d-a7ce-d65dfb2611ac)

and when removing the 34 at the beginning and end, it results in the id without the quotes. 

I then took those bytes without the quotes and converted them to a binary file and sent that binary file using postman and it worked.

**Comment by fschick** [comment]
Can you tell me where the Id is wrong? Is it a URL / query parameter? Than it should be without quotes.

Your code above `httpRequestMessage.Content = new StringContent(customJsonCodec.Serialize(options.Data), new UTF8Encoding(), "application/json");` lets me assume it's for a body with POST/PUT/... In this case, the quotes are required.
Without, it's invalid JSON

```JSON
{ "Id": bc724c54-618f-4455-9f79-382425b051c8" }
```

**Comment by fschick** [comment]
Can you post some steps to reproduce easily?

**Comment by JonasDev17** [comment]
![image](https://github.com/user-attachments/assets/dd1bd908-516f-4bf9-a96a-0f00bbd7f1a6)

This is how Keycloak expects it. No json object or anything

**Comment by JonasDev17** [comment]
The red underline is because it literally is invalid json

**Comment by fschick** [comment]
It looks like a bug in Keycloak.

Either they should change the [definition](https://www.keycloak.org/docs-api/latest/rest-api/openapi.json) to match the expected content type from `application/json` to `text/plain`:

```JSON
{
  "...": "...",
  "paths": {
    "/admin/realms/{realm}/organizations/{id}/members": {
      "get": {
        "...": "...",
      },
      "post": {
        "tags": ["Organizations"],
        "summary": "Adds the user with the specified id as a member of the organization",
        "description": "Adds, or associates, an existing user with the organization. If no user is found, or if it is already associated with the organization, an error response is returned",
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "string"
              }
            }
          }
        }
      }
    }
  }
}
```

Or they should accept a JSON-encoded string like `"95f019a8-977f-b7d5-f781-c1b2a3ef288c"`

**Comment by JonasDev17** [comment]
Ok, thanks I'll write a bug report over at their repo.

**Comment by fschick** [comment]
I think I closed the error prematurely. Waiting for a fix from Keycloak.

**Comment by fschick** [comment]
keycloak/keycloak#34512

**Comment by JonasDev17** [comment]
The PR with the fix was merged. The OpenAPI definition won't change - the fix was to strip any double quotes for the affected endpoints.

**Comment by fschick** [comment]
Thank you for the info.

There are still other issues too like keycloak/keycloak#34512 and a few I fixed [earlier](https://github.com/fschick/Keycloak.RestApiClient/blob/main/Build/openapi-generator/openapi-generator-fix-spec.js). I'd like to review them and switch the Build to [Nuke](https://nuke.build) to be able to fix such shortcomings faster. Maybe I manage to add some tests too.

The top-notch solution would be if Keycloak had tests to compare the swagger.json with theris REST API. However, my Java skills are not sufficient for a pull request here.

**Comment by JonasDev17** [comment]
> Thank you for the info.
> 
> There are still other issues too like [keycloak/keycloak#34512](https://github.com/keycloak/keycloak/issues/34512) and a few I fixed [earlier](https://github.com/fschick/Keycloak.RestApiClient/blob/main/Build/openapi-generator/openapi-generator-fix-spec.js). I'd like to review them and switch the Build to [Nuke](https://nuke.build) to be able to fix such shortcomings faster. Maybe I manage to add some tests too.
> 
> The top-notch solution would be if Keycloak had tests to compare the swagger.json with theris REST API. However, my Java skills are not sufficient for a pull request here.

The problem is they don't want any breaking changes which is understandable but thats why the problem has been solved the way it has but they said the following: 

_For the future we will discuss and pick more carefully the right content-type._

**Comment by JonasDev17** [comment]
@fschick The issue has been fixed with version 26.0.6.
